messaging/src/main/java/org/frankframework/messaging/amqp/AmqpListenerContainer.java

```diff
@@ -0,0 +1,262 @@
+import org.frankframework.receivers.ResourceLimiter;
+
+/**
+ * Run all AMQP listeners that use the same connection in a single Session, for more
+ * efficient use of system resources.
+ *
+ */
+@Log4j2
+public class AmqpListenerContainer {
+	@Autowired
+	private @Setter AmqpConnectionFactoryFactory amqpConnectionFactoryFactory;
+	private String connectionName;
+	private Connection connection;
+	private Session session;
+
+	private final Map<Receiver, AmqpListener> receivers = new ConcurrentHashMap<>();
+	private final Map<AmqpListener, Receiver> listeners = new ConcurrentHashMap<>();
+
+	private @Getter @Setter TaskExecutor taskExecutor;
+
+	private final AtomicBoolean isRunning = new AtomicBoolean(false);
+	private final AtomicBoolean isClosing = new AtomicBoolean(false);
+
+	/**
+	 * Open connection for which this instance is managing receivers. This connection needs
+	 * to be opened before any {@link AmqpListener}s can be registered with this container.
+	 *
+	 * @param connectionName
+	 * @throws LifecycleException
+	 */
+	public void openConnection(@Nonnull String connectionName) throws LifecycleException {
+		if (this.connectionName != null) {
+			throw new IllegalStateException("connectionName is already set");
+		}
+		this.connectionName = connectionName;
+
+		try {
+			connection = amqpConnectionFactoryFactory.getConnectionFactory(connectionName).getConnection();
+			SessionOptions sessionOptions = new SessionOptions();
+			session = connection.openSession(sessionOptions);
+		} catch (ClientException e) {
+			throw new LifecycleException("Cannot open AMQP connection", e);
+		}
+	}
+
+	public boolean isOpen() {
+		return connection != null;
+	}
+
+	/**
+	 * Open a listener in this listener-container. The listener-container should have an open connection
+	 * before a listener can be opened, and the listener connection-name should be for the same connection
+	 * as this listener-container instance.
+	 *
+	 * @param amqpListener
+	 */
+	public void openListener(@Nonnull AmqpListener amqpListener) {
+		if (this.connectionName == null) {
+			throw new IllegalStateException("connection for the AmqpListenerContainer is not yet open");
+		}
+		if (!this.connectionName.equals(amqpListener.getConnectionName())) {
+			throw new IllegalStateException("AmqpListener[" + amqpListener.getName() +
+					"] connection [" + amqpListener.getConnectionName() +
+					"] is not the same as connection [" + this.connectionName +
+					"] for this AmqpListenerContainer");
+		}
+		ReceiverOptions options = new ReceiverOptions();
+		options.deliveryMode(amqpListener.getDeliveryMode());
+		options.sourceOptions().capabilities(amqpListener.getAddressType().getCapabilityName());
+
+		Receiver receiver;
+		try {
+			if (amqpListener.isDurable()) {
+				receiver = session.openDurableReceiver(amqpListener.getAddress(), amqpListener.getSubscriptionName(), options);
+			} else {
+				receiver = session.openReceiver(amqpListener.getAddress(), options);
```

Comment:
Als Receiver het gene is dat je nodig hebt, en verder niets met de connecties doet, waarom geeft de factory dan niet een sessie terug?

Reply:
Daar heb ik over gedacht maar ik ben er nog niet helemaal uit hoe dan de connections zelf te managen. Wanneer op je een nieuwe, wanneer sluit je de bestaande.
Misschien dat het wel beter is inderdaad om altijd uit de factory een nieuwe session terug te geven ipv een connection, maar dat zou ik dan liever in een volgend PR doen.